packages/secure_storage/amplify_secure_storage_dart/lib/src/ffi/utils/linux_utils.dart

```diff
@@ -0,0 +1,56 @@
+import 'dart:ffi';
+import 'dart:io';
+
+import 'package:amplify_secure_storage_dart/src/ffi/gio/gio.dart';
+import 'package:ffi/ffi.dart';
+import 'package:meta/meta.dart';
+import 'package:path/path.dart' as path;
+import 'package:xdg_directories/xdg_directories.dart' as xdg;
+
+/// Returns a path to a directory where the application may place application support
+/// files.
+///
+/// Based off of path_provider.
+///
+/// Reference: https://github.com/flutter/plugins/blob/a124dcb33b620f2908aae4f28af6487163b7d8ea/packages/path_provider/path_provider_linux/lib/src/path_provider_linux.dart#L48-L67
+@internal
+Future<String> getApplicationSupportPath(String appId) async {
+  final directory = Directory(
+    path.join(
+      xdg.dataHome.path,
+      appId,
+    ),
+  );
+  if (directory.existsSync()) {
+    return directory.path;
+  }
+
+  await directory.create(recursive: true);
+  return directory.path;
```

Comment:
Generally only want to use sync APIs in like scripts or CLI tools where it's okay to block. Also DRY-d it up a bit.

```suggestion
  if (!await directory.exists()) {
    await directory.create(recursive: true);
  }
  return directory.path;
}
```

Reply:
Oh I forgot this will run in an isolate... still, probably just worth changing them to async anyway

Reply:
Yeah I will update these to use the async APIs. I based this off of shared preferences 😅 . Not sure why they use the sync APIs since the APIs they expose are all async.